blocks/main_loop.py

```diff
@@ -246,13 +247,13 @@ def _check_finish_training(self, level):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.batch_interrupt_received or
-                self.log.previous_row.batch_interrupt_received):
-            raise TrainingFinish
-        if (level == 'epoch' and
-                (self.log.current_row.epoch_interrupt_received or
-                 self.log.previous_row.epoch_interrupt_received)):
-            raise TrainingFinish
+	     self.log.current_row.batch_interrupt_received or
+	     self.log.previous_row.batch_interrupt_received):
```

Comment:
Can you please also introduce a `_batch_interrupt_received` record in the status, similarly to `_epoch_interrupt_received`? This looking into the previous row is not really nice, my bad :)